node/src/main/scala/coop/rchain/node/node.scala

```diff
@@ -89,15 +89,14 @@ class NodeRuntime(conf: Conf)(implicit scheduler: Scheduler) {
       case _ => None
     }
 
-    publicKey
-      .flatMap(CertificateHelper.publicAddress)
-      .getOrElse {
-        println("Certificate must contain a secp256r1 EC Public Key")
-        System.exit(1)
-        Array[Byte]()
-      }
-      .map("%02x".format(_))
-      .mkString
+    Base16.encode(
+      publicKey
+        .flatMap(CertificateHelper.publicAddress)
+        .getOrElse {
+          println("Certificate must contain a secp256r1 EC Public Key")
+          System.exit(1)
+          Array[Byte]()
```

Comment:
Maybe split that to the logic of extracting public key and encoding it in base16. Also I don't like the `System.exit(1)` in `getOrElse` clause. Would the default empty array from line `98` ever by returned if there's a `System.exit(1)` line before that?